### ISSUE TITLE
EPIC: breaking syscall changes

### ISSUE THREAD

**Issue by Stebalien** [filecoin-project/ref-fvm#512]
We're going to need to make a few breaking changes to syscalls, so we might as well bundle them up into a feature branch to reduce the upgrade pain:

- #506 (#509)
- #270 
- #511 
- https://github.com/filecoin-project/fvm-specs/issues/85

**Comment by raulk** [comment]
Hm, but only the first one is needed for M1 since it impacts gas
expenditures and performance of builtin actors. Agree the rest would be
nice to have, but we haven’t started yet on those and they’re not required
to M1, IMO. So it seems undesirable to synchronise all these changes.
Unless you want to do these now to not have to maintain the legacy syscalls
around forever?

**Comment by Stebalien** [comment]
> Unless you want to do these now to not have to maintain the legacy syscalls
around forever?

This. Basically, these are all easy changes as long as we:

1. Do them now.
2. Do them as one big batch.

**Comment by raulk** [comment]
Agree. Basically the policy here is: "every syscall we _have_, make sure it's in its final form ready to ossify." All issues are very well specified and are straightforward. Thinking that @vyzo could jump into (2), (3), and (4).

**Comment by Stebalien** [comment]
I'm currently working on three and, well, check slack.
